### PR TITLE
Fix: cleanup after migrations

### PR DESCRIPTION
**Pull request by nytamin**
This PR adds a check after a user has completed the migrations for removing any old data in the database.

image

* **Other information**:

**Status**

- [x] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [x] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

File: meteor/client/ui/Settings/SystemManagement.tsx [hunk 1]
```diff
@@ -282,3 +216,104 @@ export default translateWithTracker<IProps, {}, ITrackedProps>((_props: IProps)
...
+					if (result.docsToRemove > 0) {
+						affectedCollections.push(result.collectionName)
+					}
+				})
+				if (totalCount) {
+					doModalDialog({
+						title: t('Remove old data from database'),
+						message: (
+							<React.Fragment>
+								<div>
```
Comment (jstarpl) [inline meteor/client/ui/Settings/SystemManagement.tsx:263]: Should be `<p>` to make the margins work.

File: meteor/client/ui/Settings/SystemManagement.tsx [hunk 2]
```diff
@@ -282,3 +216,104 @@ export default translateWithTracker<IProps, {}, ITrackedProps>((_props: IProps)
...
+									{t('There are {{count}} documents that can be removed, do you want to continue?', {
+										count: totalCount,
+										collections: languageAnd(t, affectedCollections),
+									})}
+								</div>
+								<div>
```
Comment (jstarpl) [inline meteor/client/ui/Settings/SystemManagement.tsx:269]: This should also be a `<p>`.

File: meteor/client/ui/Settings/SystemManagement.tsx [hunk 3]
```diff
@@ -282,3 +216,104 @@ export default translateWithTracker<IProps, {}, ITrackedProps>((_props: IProps)
...
+									{t('Documents to be removed:')}
+									{collections.map((o, index) => {
```
Comment (jstarpl) [inline meteor/client/ui/Settings/SystemManagement.tsx:271]: This is a list, so it should be wrapped in a `<ul></ul>` and each item should be `<li key={index}></li>`

File: meteor/server/api/cleanup.ts [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+import { PackageInfos } from '../../lib/collections/PackageInfos'
+
+export function cleanupOldDataInner(actuallyCleanup: boolean = false): CollectionCleanupResult | string {
+	if (actuallyCleanup) {
+		const notAllowedReason = isAllowedToRunCleanup()
+		if (notAllowedReason) return `Could not run the cleanup function due to: ${notAllowedReason}`
+	}
+
+	/** Clean up stuff that are older than this: */
+	const MAXIMUM_AGE = 1000 * 60 * 60 * 24 * 100 // 100 days
```
Comment (jstarpl) [inline meteor/server/api/cleanup.ts:52]: Feels like this should be configurable through `Meteor.settings` or CoreSystem object.

**Review by jstarpl — CHANGES_REQUESTED** [review]
Very welcome addition to remove all the cruft that is leftover. Some notes on the front-end implementation.

File: meteor/server/api/cleanup.ts (hunk 4 above)
Comment (nytamin) [reply meteor/server/api/cleanup.ts:52]: Very good idea! 👍

**Pull request merged**